src/core/map/hooks/useMapPopoverMaplibreIntegration.ts

```
@@ -0,0 +1,119 @@
+import {
+  getMapContainerRect,
+  pageToMapContainerCoords,
+} from '../utils/maplibreCoordinateUtils';
+import { useMapPositionTracker } from './useMapPositionTracker';
+import type {
+  MapPopoverService,
+  ScreenPoint,
+  MapPopoverPositionCalculator,
+  IMapPopoverContentRegistry,
+} from '../types';
+import type { Map, MapMouseEvent } from 'maplibre-gl';
```

Biome (1.9.4)

[error] Do not shadow the global "Map" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)